src/display/annotation_layer.js

```diff
@@ -758,28 +792,37 @@ class PopupAnnotationElement extends AnnotationElement {
     }
 
     const selector = `[data-annotation-id="${this.data.parentId}"]`;
-    const parentElement = this.layer.querySelector(selector);
-    if (!parentElement) {
+    const parentElement = this.layer.querySelectorAll(selector);
+    if (!parentElement || parentElement.length === 0) {
```

Comment:
If it's a list now, we should call it `parentElements` (plural) for clarity. Moreover, `querySelectorAll` always returns a list, so the `!parentElement` case can no longer occur.